src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Eventing/ExceptionsEventSource.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions.Eventing
+{
+    [EventSource(Name = SourceName)]
+    internal sealed class ExceptionsEventSource : EventSource
```

Comment:
Any issues with https://github.com/dotnet/runtime/issues/76704? Should we send a marker event to ensure that all the data is flushed?

Reply:
I do need to add something that does this flushing but probably in a separate PR. I'm thinking something along the lines of a timer callback that will fire after a short time, which will be reset for every event, and finally send off the marker event when the timer expires. I'm open to other ideas too, but I don't want to always wait 100s of milliseconds after each event or exception is sent.

Reply:
Note that there aren't any tests that consume this event source from outside of the target process yet, so we don't run into the issue yet.